Catappult/appcoins-wallet-android
# [WAL-71] Integrate new gamification level up highlight

**What does this PR do?**

   Highlight of level up in gamification by sending notifications in Android. This is done via a service, instantiated when a top-up or IAP is successful. There are several possible notifications that are sent:
   1. Notification for when user reaches max level (sent once);
   2. Notification for when levels up, with/without perk bonus associated (sent once per level)
   3. Notification for when user is close to the next level, having reached a certain percentage of appcoins spent (sent once per level)
  
   Notification 3 is sent separately from notification 2, so user may receive two different gamification notifications associated to the same purchase, when user makes a large enough transaction to level up and be close to the next level.

   Read the added comments in the respective files to understand the rationale of some decisions.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] PerkBonusAndGamificationService.kt (refactored name)

**How should this be manually tested?**

  Make IAPs or Top-ups in order to level up and/or earn perk bonuses, and check if the notifications are being sent. Check notes on respective ticket: [WAL-71](https://aptoide.atlassian.net/browse/WAL-71)
  
  Regarding notification for reaching max level, you don't need to make purchases until you reach max level (will take some time). You can change one line in the code in order to change the "maximum level" that is used for comparison. You can contact me if you are not sure where this is.

**What are the relevant tickets?**

  Tickets related to this pull-request: [WAL-71](https://aptoide.atlassian.net/browse/WAL-71)

**Code Review Checklist**
